articles/root-cause-analysis.md

```diff
@@ -0,0 +1,79 @@
+---
+authorName: Matthias Meid
+authorGithubUsername: matthias-meid
+issue: 98
+title: You Always Have Time for a Proper Root Cause Analysis
+---
+# {{page.title}}
+
+A production issue will pop up, even in solid software. [Don't panic.](https://en.wikipedia.org/wiki/Phrases_from_The_Hitchhiker%27s_Guide_to_the_Galaxy#Don't_Panic)
+
+We're more than eager to resolve the issue quickly. We should be, yet we should never stop being careful and analytical. Otherwise we might deploy another embarassing bug or suppress the problem without solving it, making everything worse.
+
+Saving an hour of analysis is not worth it.
+
+## People Will Describe the Symptoms
+
+Keep in mind that users will report _symptoms_. This has nothing to do with a lack of understanding or skill. They have an outside view.
+
+What you need to find, and then fix, is the _problem_ (or more precisely its _cause_).
+
+Don't jump to conclusions. Listen to all the clues. Analyse. Re-produce. Re-think. Refine (or create) the ticket. Think of test cases which are explicit about expected and actual results.
+
+Don't have the BA do it for you, you need to be part of the thinking. This might very well take longer than the coding. But if you don't do this slowly and deliberately, you might eliminate only half the problem or even make it worse. Don't be that cowboy.
+
+## Can We Be More Specific?
+
+Absolutely!
+
+Let's have a look at some of the painfully obvious yet all too common pitfalls.
+
+### Not Really Fixing the Issue. Or: Not Fixing The Real Issue.
+
+Do you feel some guilt reading through the three examples below? I do.
+
+* Added a fix in the wrong place. Didn't solve the problem, optionally introduced another one?
+* Fixed a `NullReferenceException` by adding another `if`, ended up with no exception but a wrong result?
+* Divided an `x` through `total` to make it relative, deployed to production, got an "unexpected" `DivideByZeroException` the day after?
+
+Things are often more subtle and not that silly. What they have in common though is that we didn't think it through all the way. We knew what _shouldn't_ happen, but didn't find out _why_ and what _should_.
+
+### "It's a Feature." Or: "We Cannot Re-Produce This!"
```

Comment:
It's a Feature. -> das verstehe ich nicht ganz. Der Abschnitt scheint darauf einzugehen, aber ich weiss eingeltich gar nicht, was das heisst "It's a feature" -> den customer nicht finden ist ja kein Feature, oder?

Ich würde diesen Abschnitt ausschliesslich auf die "cannot reproduce" thematik fokussieren.